src/swsssdk/port_util.py

```diff
@@ -103,3 +108,40 @@ def get_vlan_id_from_bvid(db, bvid):
 
     return vlan_id
 
+def get_rif_port_map(db):
+    """
+        Get the RIF port mapping from ASIC DB
+    """
+    db.connect('ASIC_DB')
+    rif_keys_str = db.keys('ASIC_DB', "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:*")
+    if not rif_keys_str:
+        return {}
+
+    rif_port_oid_map = {}
+    for rif_s in rif_keys_str:
+        rif_id = rif_s.lstrip(b"ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:oid:0x")
+        ent = db.get_all('ASIC_DB', rif_s, blocking=True)
+        if b"SAI_ROUTER_INTERFACE_ATTR_PORT_ID" in ent:
+            port_id = ent[b"SAI_ROUTER_INTERFACE_ATTR_PORT_ID"].lstrip(b"oid:0x")
+            rif_port_oid_map[rif_id] = port_id
+
+    return rif_port_oid_map
+
+def get_vlan_interface_oid_map(db):
+    """
+        Get Vlan Interface names and sai oids
+    """
+    db.connect('COUNTERS_DB')
+    rif_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_RIF_NAME_MAP', blocking=True)
+    rif_type_name_map = db.get_all('COUNTERS_DB', 'COUNTERS_RIF_TYPE_MAP', blocking=True)
+
+    if not rif_name_map:
+        return {}
+
+    oid_pfx = len("oid:0x")
+    vlan_if_name_map = {sai_oid[oid_pfx:]: if_name for if_name, sai_oid in rif_name_map.items()
```

Comment:
This statement is too long to understand. Could you refactor to a loop with if-block? And add some comment to explain?

Reply:
Sure